app/uk/gov/hmrc/minorentityidentificationfrontend/services/AuditService.scala

```diff
@@ -33,76 +33,163 @@ class AuditService @Inject()(appConfig: AppConfig,
+    val retrieveCHRN = storageService.retrieveCHRN(journeyId)
+    val retrieveIdentifiersMatch = storageService.retrieveIdentifiersMatch(journeyId)
+    val retrieveBusinessVerificationStatus = storageService.retrieveBusinessVerificationStatus(journeyId)
+    val retrieveRegistrationStatus = storageService.retrieveRegistrationStatus(journeyId)
+
+    for {
+      optSaUtr <- retrieveSaUtr
+      optSaPostCode <- retrieveSaPostCode
+      optCHRN <- retrieveCHRN
+      optIdentifiersMatch <- retrieveIdentifiersMatch
+      optBusinessVerificationStatus <- retrieveBusinessVerificationStatus
+      optRegistrationStatus <- retrieveRegistrationStatus
+    } yield {
+
+      val optSaUtrBlock = optSaUtr match {
+        case Some(saUtr) => Json.obj("SAUTR" -> saUtr.value)
+        case None => Json.obj()
+      }
+
+      val optSaPostCodeBlock = optSaPostCode match {
+        case Some(saPostCode) => Json.obj("SApostcode" -> saPostCode)
+        case None => Json.obj()
       }
+
+      val optCHRNBlock = optCHRN match {
+        case Some(chrn) => Json.obj("CHRN" -> chrn)
+        case None => Json.obj()
+      }
+
+      val auditJson = Json.obj(
+        "callingService" -> callingService,
+        "businessType" -> "Trusts",
+        "isMatch" -> defineAuditIdentifiersMatch(optIdentifiersMatch),
+        "VerificationStatus" -> defineAuditBusinessVerificationStatus(optBusinessVerificationStatus),
+        "RegisterApiStatus" -> defineAuditRegistrationStatus(optRegistrationStatus)
+      ) ++ optSaUtrBlock ++ optSaPostCodeBlock ++ optCHRNBlock
+
+      auditConnector.sendExplicitAudit(
+        auditType = "TrustsRegistration",
+        detail = auditJson
+      )
+    }
+  }
+
+  private def auditUnincorporatedAssociationJourney(journeyId: String, callingService: String)
+                               (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Unit] = {
+
+    for {
+      optRegistrationStatus <- storageService.retrieveRegistrationStatus(journeyId)
+    } yield {
+      val auditJson = Json.obj(
+        "callingService" -> callingService,
+        "businessType" -> "Unincorporated Association",
+        "identifiersMatch" -> false,
+        "VerificationStatus" -> Json.obj("verificationStatus" -> "UNCHALLENGED"),
+        "RegisterApiStatus" -> defineAuditRegistrationStatus(optRegistrationStatus)
+      )
+
+      auditConnector.sendExplicitAudit(
+        auditType = "UnincorporatedAssociationRegistration",
+        detail = auditJson
+      )
     }
   }
+
+  private def defineAuditIdentifiersMatch(optIdentifiersMatch: Option[KnownFactsMatchingResult]): String =
+    optIdentifiersMatch match {
+      case Some(SuccessfulMatch) => "true"
+      case Some(DetailsMismatch) | Some(DetailsNotFound) => "false"
+      case Some(UnMatchableWithRetry) | Some(UnMatchableWithoutRetry) => "unmatchable"
+      case None => "false" // TODO Should we default to "false"
```

Comment:
Actually, we do need this line at present, because the code can run with or without the feature switch enabled. If the switch is disabled there will be no entry for identifiers match.